File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) < 2:
            return 0
        
        max_profit = 0
        min_price = prices[0]

        for _, price in enumerate(prices, 1):
            current_profit = price - min_price
            max_profit = max(max_profit, current_profit)
            min_price = min(min_price, price)
        
        return max_profit
            

        
        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    if(prices.length === 1 )
        return 0

    let lp = prices[0], maxP = 0;

    for( let i=1; i < prices.length; i++ ) {
        let profit = prices[i] - lp;

        lp = Math.min(lp, prices[i]);
        maxP = Math.max(maxP, profit);
    }

    return maxP
};